connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/protobuf/functions.scala

```diff
@@ -45,12 +53,36 @@ object functions {
       messageName: String,
       descFilePath: String,
       options: java.util.Map[String, String]): Column = {
+    val binaryFileDescSet = readDescriptorFileContent(descFilePath)
```

Comment:
@LuciferYang I updated connect functions to use the buffer and change the protocol to send the buffer to the server rather the file name (as a result the file is only read on the client side). 
Updated the tests and added a ClientE2E test. PTAL.

Reply:
Sorry for the late reply, will review this one later

Reply:
@LuciferYang PTAL when you get a chance. I think the changes are fairly straight forward.